tests/functional/test_supported_protocols.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import pytest
+
+from botocore.args import PRIORITY_ORDERED_SUPPORTED_PROTOCOLS
+from botocore.session import get_session
+
+
+def _all_test_cases():
+    session = get_session()
+    loader = session.get_component('data_loader')
+
+    services = loader.list_available_services('service-2')
+    services_models_with_protocols = []
+    services_models_without_protocols = []
+
+    for service in services:
+        service_model = session.get_service_model(service)
+        if 'protocols' in service_model.metadata:
+            services_models_with_protocols.append(service_model)
+        else:
+            services_models_without_protocols.append(service_model)
+    return (services_models_with_protocols, services_models_without_protocols)
```

Comment:
[Not Blocking]This seems pretty memory inefficient as opposed to using `yield`. I don't feel strongly though given this pattern is used in other tests and this isn't actually SDK code.

Reply:
I've updated it to something "better"?  It doesn't load them all into memory, but it feels a bit awkward.  I'd love some feedback on it when you get the chance.